File: recursion.scala
// recursion.scala
object RecursionExample {
  def factorial(n: Int): Int = {
    if (n <= 1) 1
    else n * factorial(n - 1)
  }

  def main(args: Array[String]): Unit = {
    print("Enter an integer: ")
    val input = scala.io.StdIn.readLine()
    val param = input.toInt
    if (param == 0 && input != "0") {
      println("Invalid input")
    } else {
      println("Factorial: " + factorial(param))
    }
  }
}

File: recursion.pas
{ recursion.pas }
program RecursionExample;

function Factorial(Param: Integer): Integer;
begin
    if Param <= 1 then
        Factorial := 1
    else
        Factorial := Param * Factorial(Param - 1);
end;

var
    Input: string;
    Param: Integer;

begin
    Write('Enter an integer: ');
    Readln(Input);
    Val(Input, Param);
    if (Param = 0) and (Input <> '0') then
        Writeln('Invalid input')
    else
        Writeln('Factorial: ', Factorial(Param));
end.
